actors/builtin/miner/miner_actor.go

```diff
@@ -2067,6 +2068,253 @@ func (a Actor) RepayDebt(rt Runtime, _ *abi.EmptyValue) *abi.EmptyValue {
+		sectorsDataSpec = append(sectorsDataSpec, &market.SectorDataSpec{
+			SectorType: sectorInfo.SealProof,
+			DealIDs:    update.Deals,
+		})
+	}
+
+	//// TODO: this method has some potential aborts -- are those all programmer error or is there scope to improve UX for bad input?
+	dealWeights := requestDealWeights(rt, sectorsDeals)
+
+	builtin.RequirePredicate(rt, len(dealWeights.Sectors) == len(prevalidatedUpdates), exitcode.ErrIllegalState,
+		"deal weight request returned %d records, expected %d", len(dealWeights.Sectors), len(prevalidatedUpdates))
+
+	// TODO: this method has some potential aborts -- are those all programmer error or is there scope to improve UX for bad input?
+	unsealedSectorCIDs := requestUnsealedSectorCIDs(rt, sectorsDataSpec...)
+
+	builtin.RequirePredicate(rt, len(unsealedSectorCIDs) == len(prevalidatedUpdates), exitcode.ErrIllegalState,
+		"unsealed sector cid request returned %d records, expected %d", len(unsealedSectorCIDs), len(prevalidatedUpdates))
+
+	succeededSectors := bitfield.New()
+	var st State
+	rt.StateTransaction(&st, func() {
+		for k, updateWithSectorInfo := range prevalidatedUpdates {
+			err = rt.VerifyReplicaUpdate(
+				proof.ReplicaUpdateInfo{
+					NewSealedSectorCID:   updateWithSectorInfo.update.NewSealedSectorCID,
+					OldSealedSectorCID:   updateWithSectorInfo.sectorInfo.SealedCID,
+					NewUnsealedSectorCID: unsealedSectorCIDs[k],
+				})
+
+			if err != nil {
+				rt.Log(rtt.INFO, "failed to verify replica proof, skipping sector %d", updateWithSectorInfo.sectorInfo.SectorNumber)
+				continue
+			}
+
+			newSectorInfo := updateWithSectorInfo.sectorInfo
+
+			newSectorInfo.SealedCID = updateWithSectorInfo.update.NewSealedSectorCID
+			newSectorInfo.DealIDs = updateWithSectorInfo.update.Deals
+
+			newSectorInfo.DealWeight = dealWeights.Sectors[k].DealWeight
+			newSectorInfo.VerifiedDealWeight = dealWeights.Sectors[0].VerifiedDealWeight
+
+			// compute initial pledge
+			duration := updateWithSectorInfo.sectorInfo.Expiration - rt.CurrEpoch()
+
+			// TODO: Should we include this?
+			//if duration < MinSectorExpiration {
+			//	rt.Log(rtt.WARN, "precommit %d has lifetime %d less than minimum. ignoring", precommit.Info.SectorNumber, duration, MinSectorExpiration)
+			//	continue
+			//}
+
+			pwr := QAPowerForWeight(info.SectorSize, duration, newSectorInfo.DealWeight, newSectorInfo.VerifiedDealWeight)
+
+			newSectorInfo.ReplacedDayReward = updateWithSectorInfo.sectorInfo.ExpectedDayReward
+			newSectorInfo.ExpectedDayReward = ExpectedRewardForPower(rewRet.ThisEpochRewardSmoothed, powRet.QualityAdjPowerSmoothed, pwr, builtin.EpochsInDay)
+			newSectorInfo.ExpectedStoragePledge = ExpectedRewardForPower(rewRet.ThisEpochRewardSmoothed, powRet.QualityAdjPowerSmoothed, pwr, InitialPledgeProjectionPeriod)
+			newSectorInfo.ReplacedSectorAge = maxEpoch(0, rt.CurrEpoch()-updateWithSectorInfo.sectorInfo.Activation)
```

Comment:
The new sector info looks more or less good but it is currently not being set.  You'll need to set a reference to grab a reference to the sectors AMT (I think your `sectors` variable should be fine) and modify it to set this sector info at the sectorID index in the AMT.  Then you'll need to flush this AMT at the very end before leaving the StateTransaction.